Lib/typing.py

```diff
@@ -2001,6 +1997,14 @@ def _allow_reckless_class_checks(depth=3):
 class _ProtocolMeta(ABCMeta):
     # This metaclass is really unfortunate and exists only because of
     # the lack of __instancehook__.
+    def __init__(cls, *args, **kwargs):
+        cls.__protocol_attrs__ = _get_protocol_attrs(cls)
+        # PEP 544 prohibits using issubclass()
+        # with protocols that have non-method members.
+        cls.__callable_proto_members_only__ = all(
+            callable(getattr(cls, attr, None)) for attr in cls.__protocol_attrs__
+        )
```

Comment:
The only reason I'm adding this method here rather than doing this work in `Protocol.__init_subclass__` is that it seems slightly more backwards-compatible. With this PR, `_ProtcolMeta.__instancecheck__` assumes that all classes with `_ProtocolMeta` as their metaclass will have a `__protocol_attrs__` attribute. There _should_ only be one class using `_ProtocolMeta` as its metaclass (`Protocol`), and if we could count on that, then it would be safe to do this work in `Protocol.__init_subclass__`. But it's possible users might have been reaching into the internals of `typing.py` and creating other classes that use `_ProtocolMeta` as their metaclass, and I don't want to risk breaking their code unnecessarily.

Reply:
Confirmed that there's at least a few uses of `_ProtocolMeta` out in the wild:

https://github.com/antonagestam/phantom-types/blob/68ee857af4452a90a48669a8dbc7d1d7110b31d0/src/phantom/sized.py#L71-L82

https://github.com/protolambda/remerkleable/blob/91ed092d08ef0ba5ab076f0a34b0b371623db728/remerkleable/core.py#L18-L22

https://github.com/InterStella0/stella_bot/blob/bf5f5632bcd88670df90be67b888c282c6e83d99/utils/useful.py#L337-L343